# Homeless Concerns

In [1]:
import pandas as pd 
import numpy as np

In [2]:
df = pd.read_csv('../../data/raw/Homeless_Concerns.csv')

In [3]:
df.head()

CaseID                  Opened                  Closed  \
0  11770932  12/04/2019 03:47:00 PM                     NaN   
1  11719142  11/21/2019 01:14:00 PM  11/21/2019 02:08:41 PM   
2  11669592  11/10/2019 10:44:00 AM  11/10/2019 10:56:28 AM   
3  11669528  11/10/2019 10:31:00 AM  11/10/2019 10:56:42 AM   
4  11640478  11/03/2019 04:19:00 PM  11/03/2019 05:08:52 PM   

                  Updated  Status  \
0  12/05/2019 03:56:44 AM    Open   
1  11/21/2019 02:08:41 PM  Closed   
2  11/10/2019 10:56:28 AM  Closed   
3  11/10/2019 10:56:42 AM  Closed   
4  11/03/2019 05:08:52 PM  Closed   

                                        Status Notes  \
0                                             closed   
1  This issue has already been reported through a...   
2                    See CAD# 193141269 for details.   
3                    See CAD# 193141258 for details.   
4                    See CAD# 193072861 for details.   

                 Responsible Agency           Category         Request Type  \
0  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   
1  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   
2  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   
3  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   
4  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   

       Request Details                                      Address  \
0       homeless_other  759 CLEMENTINA ST, SAN FRANCISCO, CA, 94103   
1      wellbeing_check  1414 VAN NESS AVE, SAN FRANCISCO, CA, 94109   
2  aggressive_behavior   Intersection of CHURCH ST and RESERVOIR ST   
3       homeless_other       617 ELLIS ST, SAN FRANCISCO, CA, 94109   
4      wellbeing_check       3172 22ND ST, SAN FRANCISCO, CA, 94110   

          Street  Supervisor District     Neighborhood Police District  \
0  CLEMENTINA ST                  6.0  South of Market        SOUTHERN   
1   VAN NESS AVE                  3.0   Lower Nob Hill        NORTHERN   
2      CHURCH ST                  8.0  Duboce Triangle            PARK   
3       ELLIS ST                  6.0       Tenderloin      TENDERLOIN   
4        22ND ST                  9.0          Mission         MISSION   

    Latitude   Longitude                         Point          Source  \
0  37.774707 -122.411510  (37.77470701, -122.41150998)  Mobile/Open311   
1  37.788818 -122.421570  (37.78881836, -122.42156982)           Phone   
2  37.768701 -122.428627  (37.76870057, -122.42862669)  Mobile/Open311   
3  37.784306 -122.416569  (37.78430598, -122.41656898)  Mobile/Open311   
4  37.755469 -122.418176  (37.75546882, -122.41817633)  Mobile/Open311   

   Media URL  
0        NaN  
1        NaN  
2        NaN  
3        NaN  
4        NaN

In [4]:
df.dtypes

CaseID                   int64
Opened                  object
Closed                  object
Updated                 object
Status                  object
Status Notes            object
Responsible Agency      object
Category                object
Request Type            object
Request Details         object
Address                 object
Street                  object
Supervisor District    float64
Neighborhood            object
Police District         object
Latitude               float64
Longitude              float64
Point                   object
Source                  object
Media URL              float64
dtype: object

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29128 entries, 0 to 29127
Data columns (total 20 columns):
CaseID                 29128 non-null int64
Opened                 29128 non-null object
Closed                 29116 non-null object
Updated                29128 non-null object
Status                 29128 non-null object
Status Notes           28113 non-null object
Responsible Agency     29128 non-null object
Category               29128 non-null object
Request Type           29128 non-null object
Request Details        29128 non-null object
Address                29128 non-null object
Street                 29095 non-null object
Supervisor District    29095 non-null float64
Neighborhood           28910 non-null object
Police District        28899 non-null object
Latitude               29128 non-null float64
Longitude              29128 non-null float64
Point                  29128 non-null object
Source                 29128 non-null object
Media URL              0 non-null 

## Rename columns

In [6]:
df.rename(columns={'Latitude': 'Y',
                  'Longitude': 'X'}, inplace=True)
df.head()

CaseID                  Opened                  Closed  \
0  11770932  12/04/2019 03:47:00 PM                     NaN   
1  11719142  11/21/2019 01:14:00 PM  11/21/2019 02:08:41 PM   
2  11669592  11/10/2019 10:44:00 AM  11/10/2019 10:56:28 AM   
3  11669528  11/10/2019 10:31:00 AM  11/10/2019 10:56:42 AM   
4  11640478  11/03/2019 04:19:00 PM  11/03/2019 05:08:52 PM   

                  Updated  Status  \
0  12/05/2019 03:56:44 AM    Open   
1  11/21/2019 02:08:41 PM  Closed   
2  11/10/2019 10:56:28 AM  Closed   
3  11/10/2019 10:56:42 AM  Closed   
4  11/03/2019 05:08:52 PM  Closed   

                                        Status Notes  \
0                                             closed   
1  This issue has already been reported through a...   
2                    See CAD# 193141269 for details.   
3                    See CAD# 193141258 for details.   
4                    See CAD# 193072861 for details.   

                 Responsible Agency           Category         Request Type  \
0  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   
1  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   
2  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   
3  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   
4  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   

       Request Details                                      Address  \
0       homeless_other  759 CLEMENTINA ST, SAN FRANCISCO, CA, 94103   
1      wellbeing_check  1414 VAN NESS AVE, SAN FRANCISCO, CA, 94109   
2  aggressive_behavior   Intersection of CHURCH ST and RESERVOIR ST   
3       homeless_other       617 ELLIS ST, SAN FRANCISCO, CA, 94109   
4      wellbeing_check       3172 22ND ST, SAN FRANCISCO, CA, 94110   

          Street  Supervisor District     Neighborhood Police District  \
0  CLEMENTINA ST                  6.0  South of Market        SOUTHERN   
1   VAN NESS AVE                  3.0   Lower Nob Hill        NORTHERN   
2      CHURCH ST                  8.0  Duboce Triangle            PARK   
3       ELLIS ST                  6.0       Tenderloin      TENDERLOIN   
4        22ND ST                  9.0          Mission         MISSION   

           Y           X                         Point          Source  \
0  37.774707 -122.411510  (37.77470701, -122.41150998)  Mobile/Open311   
1  37.788818 -122.421570  (37.78881836, -122.42156982)           Phone   
2  37.768701 -122.428627  (37.76870057, -122.42862669)  Mobile/Open311   
3  37.784306 -122.416569  (37.78430598, -122.41656898)  Mobile/Open311   
4  37.755469 -122.418176  (37.75546882, -122.41817633)  Mobile/Open311   

   Media URL  
0        NaN  
1        NaN  
2        NaN  
3        NaN  
4        NaN

## Empty values

In [7]:
for col in df:
    print(col, df[col].isnull().sum() )

CaseID 0
Opened 0
Closed 12
Updated 0
Status 0
Status Notes 1015
Responsible Agency 0
Category 0
Request Type 0
Request Details 0
Address 0
Street 33
Supervisor District 33
Neighborhood 218
Police District 229
Y 0
X 0
Point 0
Source 0
Media URL 29128


Closed nan values are fine, because the case is still open and we will only look at the open date. Status notes and request type are also not useful. Basically we need to take care of nan values for Location, Supervisor District, Neighborhood and Police District

In [8]:
df[df['Supervisor District'].isnull()]

CaseID                  Opened                  Closed  \
17      8957722  05/04/2018 10:35:00 AM  05/04/2018 11:48:32 AM   
19      8031356  09/18/2017 06:58:00 AM  09/18/2017 12:06:03 AM   
178     8760189  03/19/2018 09:13:00 AM  03/20/2018 10:22:10 AM   
2655    5457852  01/12/2016 01:05:19 PM  01/12/2016 10:46:50 PM   
8337    6518551  11/10/2016 02:37:02 PM  11/11/2016 07:44:32 AM   
9304    6514358  11/09/2016 01:20:43 PM  11/09/2016 08:38:14 PM   
10256   6862771  02/21/2017 09:59:18 AM  02/21/2017 12:43:29 PM   
10371   6862831  02/21/2017 10:06:14 AM  02/21/2017 11:23:46 AM   
13083   7530645  08/21/2017 09:39:01 AM  08/21/2017 11:35:09 AM   
14482   7387506  07/18/2017 09:12:39 AM  07/18/2017 11:58:10 AM   
14509   7363640  07/12/2017 08:19:07 AM  07/12/2017 09:07:45 AM   
14626   7475646  08/08/2017 10:20:54 AM  08/08/2017 11:33:31 AM   
15201   7397693  07/20/2017 10:35:30 AM  07/20/2017 11:54:23 AM   
15599   7344918  07/07/2017 09:01:18 AM  07/07/2017 11:02:11 AM   
16583   7529899  08/21/2017 06:54:34 AM  08/21/2017 10:00:36 AM   
17371   8054485  09/22/2017 05:44:00 PM  02/08/2019 01:38:00 PM   
17782   8015301  09/13/2017 03:33:00 PM  09/14/2017 04:48:05 AM   
17962   8051362  09/22/2017 08:33:00 AM  09/22/2017 05:17:03 AM   
18307   8194178  10/27/2017 12:24:00 PM  10/27/2017 11:41:23 AM   
18383   8017191  09/14/2017 08:23:00 AM  09/16/2017 01:38:07 AM   
18535   8036470  09/19/2017 07:52:00 AM  09/19/2017 01:00:39 AM   
19424   8103043  10/05/2017 06:58:00 AM  10/05/2017 08:39:58 PM   
19805   8070684  09/27/2017 09:01:00 AM  09/27/2017 03:01:00 AM   
19889   8182788  10/25/2017 07:28:00 AM  10/25/2017 05:06:33 AM   
20095   8041261  09/20/2017 07:35:00 AM  09/20/2017 01:10:53 AM   
20250   7578466  09/01/2017 06:51:45 AM  09/01/2017 07:57:37 AM   
20768   8341928  12/04/2017 05:14:00 AM  12/04/2017 05:34:39 AM   
21514   8660115  02/21/2018 10:45:00 AM  02/21/2018 11:45:45 AM   
21739   8404457  12/19/2017 10:21:00 AM  12/19/2017 11:45:40 AM   
23870   8787878  03/26/2018 07:56:00 AM  03/26/2018 10:32:23 PM   
26757   8715047  03/07/2018 06:53:00 AM  03/07/2018 10:15:54 PM   
29016  10365566  01/16/2019 03:54:00 PM  01/18/2019 06:27:00 AM   
29043  10647171  03/26/2019 09:00:00 AM  04/08/2019 02:16:00 PM   

                      Updated  Status  \
17     05/04/2018 11:48:32 AM  Closed   
19     09/18/2017 12:06:03 AM  Closed   
178    03/20/2018 10:22:10 AM  Closed   
2655   01/12/2016 10:46:50 PM  Closed   
8337   11/11/2016 07:44:32 AM  Closed   
9304   11/09/2016 08:38:14 PM  Closed   
10256  02/21/2017 12:43:29 PM  Closed   
10371  02/21/2017 11:23:46 AM  Closed   
13083  08/21/2017 11:35:09 AM  Closed   
14482  07/18/2017 11:58:10 AM  Closed   
14509  07/12/2017 09:07:45 AM  Closed   
14626  08/08/2017 11:33:31 AM  Closed   
15201  07/20/2017 11:54:23 AM  Closed   
15599  07/07/2017 11:02:11 AM  Closed   
16583  08/21/2017 10:00:36 AM  Closed   
17371  02/08/2019 01:38:00 PM  Closed   
17782  09/14/2017 04:48:05 AM  Closed   
17962  09/22/2017 05:17:03 AM  Closed   
18307  10/27/2017 11:41:23 AM  Closed   
18383  09/16/2017 01:38:07 AM  Closed   
18535  09/19/2017 01:00:39 AM  Closed   
19424  10/05/2017 08:39:58 PM  Closed   
19805  09/27/2017 03:01:00 AM  Closed   
19889  10/25/2017 05:06:33 AM  Closed   
20095  09/20/2017 01:10:53 AM  Closed   
20250  09/01/2017 07:57:37 AM  Closed   
20768  12/04/2017 05:34:39 AM  Closed   
21514  02/21/2018 11:45:45 AM  Closed   
21739  12/19/2017 11:45:40 AM  Closed   
23870  03/26/2018 10:32:23 PM  Closed   
26757  03/07/2018 10:15:54 PM  Closed   
29016  01/18/2019 06:27:00 AM  Closed   
29043  04/08/2019 02:16:00 PM  Closed   

                                            Status Notes  \
17     Insufficient infomation provided for action to...   
19     Insufficient infomation provided for action to...   
178    Police Officer responded to request under CAD ...   
2655   Police CAD# 160123464 assigned; Call Closed: 2...   
8337   Police Officer responded to 

We see a problem. The rows with no district are the ones with location (0,0). Since we are going to work based on locations. Drop all the locations with (0,0)

In [9]:
df = df[df.Point != '(0.0, 0.0)']
df[df['Supervisor District'].isnull()]

CaseID                  Opened                  Closed  \
2655   5457852  01/12/2016 01:05:19 PM  01/12/2016 10:46:50 PM   
13083  7530645  08/21/2017 09:39:01 AM  08/21/2017 11:35:09 AM   
14482  7387506  07/18/2017 09:12:39 AM  07/18/2017 11:58:10 AM   
14509  7363640  07/12/2017 08:19:07 AM  07/12/2017 09:07:45 AM   
15201  7397693  07/20/2017 10:35:30 AM  07/20/2017 11:54:23 AM   
15599  7344918  07/07/2017 09:01:18 AM  07/07/2017 11:02:11 AM   
16583  7529899  08/21/2017 06:54:34 AM  08/21/2017 10:00:36 AM   
17782  8015301  09/13/2017 03:33:00 PM  09/14/2017 04:48:05 AM   
17962  8051362  09/22/2017 08:33:00 AM  09/22/2017 05:17:03 AM   
20768  8341928  12/04/2017 05:14:00 AM  12/04/2017 05:34:39 AM   
21514  8660115  02/21/2018 10:45:00 AM  02/21/2018 11:45:45 AM   
21739  8404457  12/19/2017 10:21:00 AM  12/19/2017 11:45:40 AM   

                      Updated  Status  \
2655   01/12/2016 10:46:50 PM  Closed   
13083  08/21/2017 11:35:09 AM  Closed   
14482  07/18/2017 11:58:10 AM  Closed   
14509  07/12/2017 09:07:45 AM  Closed   
15201  07/20/2017 11:54:23 AM  Closed   
15599  07/07/2017 11:02:11 AM  Closed   
16583  08/21/2017 10:00:36 AM  Closed   
17782  09/14/2017 04:48:05 AM  Closed   
17962  09/22/2017 05:17:03 AM  Closed   
20768  12/04/2017 05:34:39 AM  Closed   
21514  02/21/2018 11:45:45 AM  Closed   
21739  12/19/2017 11:45:40 AM  Closed   

                                            Status Notes  \
2655   Police CAD# 160123464 assigned; Call Closed: 2...   
13083  Police Officer responded to request under CAD ...   
14482  Police Officer responded to request under CAD ...   
14509  Police Officer responded to request under CAD ...   
15201  Police Officer responded to request under CAD ...   
15599  Police Officer responded to request under CAD ...   
16583  Police Officer responded to request under CAD ...   
17782  Police Officer responded to request under CAD ...   
17962  Police Officer responded to request under CAD ...   
20768       Unknown CAD Disposition, Default to Resolved   
21514  Police Officer responded to request under CAD ...   
21739  Insufficient infomation provided for action to...   

                     Responsible Agency           Category  \
2655   Police - Homeless Concerns Queue  Homeless Concerns   
13083  Police - Homeless Concerns Queue  Homeless Concerns   
14482  Police - Homeless Concerns Queue  Homeless Concerns   
14509  Police - Homeless Concerns Queue  Homeless Concerns   
15201  Police - Homeless Concerns Queue  Homeless Concerns   
15599  Police - Homeless Concerns Queue  Homeless Concerns   
16583  Police - Homeless Concerns Queue  Homeless Concerns   
17782  Police - Homeless Concerns Queue  Homeless Concerns   
17962  Police - Homeless Concerns Queue  Homeless Concerns   
20768  Police - Homeless Concerns Queue  Homeless Concerns   
21514  Police - Homeless Concerns Queue  Homeless Concerns   
21739  Police - Homeless Concerns Queue  Homeless Concerns   

              Request Type      Request Details  \
2655   Individual Concerns  aggressive_behavior   
13083  Individual Concerns       homeless_other   
14482  Individual Concerns       homeless_other   
14509  Individual Concerns       homeless_other   
15201  Individual Concerns       homeless_other   
15599  Individual Concerns       homeless_other   
16583  Individual Concerns       homeless_other   
17782  Individual Concerns       homeless_other   
17962  Individual Concerns       homeless_other   
20768  Individual Concerns      wellbeing_check   
21514  Individual Concerns  aggressive_behavior   
21739  Individual Concerns  aggressive_behavior   

                                      Address Street  Supervisor District  \
2655                                Tara Boss    NaN                  NaN   
13083                               Tara Boss    NaN                  NaN   
14482                               Tara Boss    NaN                  NaN   
14509                               Tara Boss    NaN   

We got rid from a lot of missing location rows, but we are still missing some supervisor districts. We can get the correct district using the location later with our class.

In [10]:
print('Nan on Neighborhood', df[df['Neighborhood'].isnull()].shape[0])
print('Nan on Police District', df[df['Police District'].isnull()].shape[0])

Nan on Neighborhood 197
Nan on Police District 208


In [11]:
df[df['Neighborhood'].isnull()]

CaseID                  Opened                  Closed  \
368    5219349  10/23/2015 04:49:13 PM  10/23/2015 10:51:22 PM   
500    5227737  10/27/2015 08:17:46 AM  10/27/2015 10:13:40 AM   
653    5220947  10/24/2015 01:15:02 PM  10/24/2015 08:41:18 PM   
674    5227730  10/27/2015 08:17:05 AM  10/27/2015 10:13:42 AM   
890    5589336  02/21/2016 03:41:02 PM  02/22/2016 03:22:37 PM   
2655   5457852  01/12/2016 01:05:19 PM  01/12/2016 10:46:50 PM   
3882   5784918  04/21/2016 08:10:55 AM  04/21/2016 08:42:50 AM   
4018   5683904  03/20/2016 10:38:51 PM  03/21/2016 06:39:04 AM   
4069   5970137  06/11/2016 05:36:34 PM  06/13/2016 07:31:38 AM   
4148   5994816  06/19/2016 12:07:15 AM  06/20/2016 10:33:51 AM   
4337   5918497  05/28/2016 12:54:55 PM  05/29/2016 06:04:45 AM   
4508   5683908  03/20/2016 10:39:41 PM  03/21/2016 06:53:23 AM   
4586   5964181  06/09/2016 06:35:53 PM  06/10/2016 08:22:04 AM   
4858   5761810  04/14/2016 08:17:53 AM  04/14/2016 09:13:20 AM   
4910   5964210  06/09/2016 06:49:33 PM  06/10/2016 08:27:10 AM   
5158   5953987  06/07/2016 12:33:00 PM  06/07/2016 09:26:15 PM   
5192   5817407  04/30/2016 05:56:42 PM  05/01/2016 07:13:28 AM   
5205   5683901  03/20/2016 10:35:15 PM  03/21/2016 06:53:36 AM   
5425   6387528  10/05/2016 03:51:37 PM  10/05/2016 04:21:22 PM   
6081   6049072  07/04/2016 08:11:21 PM  07/05/2016 11:54:03 AM   
6195   6249084  08/29/2016 09:23:52 AM  08/29/2016 10:19:41 AM   
6245   6217285  08/20/2016 09:40:52 AM  08/20/2016 11:02:03 AM   
6250   6358532  09/28/2016 12:33:45 AM  09/28/2016 01:09:50 AM   
6546   6100499  07/20/2016 08:42:36 AM  07/25/2016 03:10:59 PM   
7285   6361785  09/28/2016 04:28:54 PM  09/28/2016 06:41:44 PM   
7339   6315188  09/15/2016 02:45:19 PM  09/15/2016 03:46:37 PM   
7377   6044516  07/02/2016 07:03:22 PM  07/03/2016 07:00:14 PM   
7759   6514672  11/09/2016 02:33:55 PM  11/09/2016 03:43:22 PM   
7810   6807496  02/04/2017 05:13:37 PM  02/04/2017 06:24:45 PM   
7994   6663475  12/26/2016 10:59:37 PM  12/26/2016 11:19:01 PM   
...        ...                     ...                     ...   
18185  8007432  09/12/2017 03:31:00 AM  09/11/2017 09:33:19 PM   
18212  7569403  08/30/2017 09:33:58 AM  08/30/2017 11:06:15 AM   
18856  8074885  09/28/2017 07:31:00 AM  09/28/2017 02:10:33 AM   
19148  7608065  09/08/2017 07:53:44 AM  09/08/2017 09:56:18 AM   
19294  8083271  09/30/2017 01:17:00 AM  09/29/2017 08:23:20 PM   
19334  8286089  11/18/2017 08:27:00 PM  11/18/2017 09:03:17 PM   
19480  8240728  11/08/2017 01:09:00 AM  11/08/2017 02:23:00 AM   
19614  8123022  10/10/2017 10:08:00 AM  10/10/2017 10:35:00 AM   
19740  8259382  11/12/2017 04:50:00 PM  11/21/2017 05:40:00 AM   
19764  8097954  10/04/2017 01:07:00 AM  10/03/2017 08:03:18 PM   
19773  7568701  08/30/2017 07:32:22 AM  08/30/2017 09:47:02 AM   
19892  8216260  11/01/2017 11:10:00 PM  11/01/2017 11:47:55 PM   
20038  8107633  10/05/2017 10:30:00 PM  10/05/2017 03:50:47 PM   
20045  8182834  10/25/2017 07:40:00 AM  10/25/2017 01:37:20 AM   
20297  8368657  12/10/2017 12:52:00 AM  12/10/2017 04:10:21 AM   
20381  8354693  12/06/2017 03:56:00 PM  12/06/2017 04:10:55 PM   
20768  8341928  12/04/2017 05:14:00 AM  12/04/2017 05:34:39 AM   
20795  8416163  12/22/2017 04:08:00 AM  12/22/2017 06:02:04 AM   
20843  8587679  02/03/2018 08:34:00 AM  02/03/2018 09:12:29 AM   
21091  8365638  12/09/2017 12:37:00 AM  12/09/2017 12:56:51 AM   
21514  8660115  02/21/2018 10:45:00 AM  02/21/2018 11:45:45 AM   
21622  8320367  11/28/2017 02:01:00 PM  11/28/2017 03:37:35 PM   
21739  8404457  12/19/2017 10:21:00 AM  12/19/2017 11:45:40 AM   
21832  8587695  02/03/2018 08:39:00 AM  02/03/2018 09:10:39 AM   
21914  8511547  01/16/2018 10:22:00 AM  01/16/2018 12:06:02 PM   
22058  8380985  12/13/2017 07:34:00 AM  12/13/2017 08:34:52 AM   
22256  8493311  01/11/2018 01:03:00 PM  01/11/2018 11:03:01 PM   
22501  8532494  01/21/2018 12:22:00 AM  01/21/2018 03:41:14 AM   
23419  8509141  01/15/2018 07:37:00 PM  01/15/20

There are less than 208 rows with nan out of 29128, that is 0.7%. Therefore, we will drop them without risk of losing data

In [12]:
df.dropna(subset=['Neighborhood'], inplace=True)
df.dropna(subset=['Police District'], inplace=True)
df.dropna(subset=['Supervisor District'], inplace=True)

In [13]:
print('Nan on Neighborhood', df[df['Neighborhood'].isnull()].shape[0])
print('Nan on Police District', df[df['Police District'].isnull()].shape[0])
print('Nan on Supervisor District', df[df['Supervisor District'].isnull()].shape[0])

Nan on Neighborhood 0
Nan on Police District 0
Nan on Supervisor District 0


## Datetime objects

In [14]:
df['Opened'] = pd.to_datetime(df['Opened'] )
df['Closed'] = pd.to_datetime(df['Closed'] )
df['Updated'] = pd.to_datetime(df['Updated'] )

In [15]:
df['Year'] = df['Opened'].dt.year
df['Month'] = df['Opened'].dt.month
df['Day'] = df['Opened'].dt.day

In [16]:
df.head()

CaseID              Opened              Closed             Updated  \
0  11770932 2019-12-04 15:47:00                 NaT 2019-12-05 03:56:44   
1  11719142 2019-11-21 13:14:00 2019-11-21 14:08:41 2019-11-21 14:08:41   
2  11669592 2019-11-10 10:44:00 2019-11-10 10:56:28 2019-11-10 10:56:28   
3  11669528 2019-11-10 10:31:00 2019-11-10 10:56:42 2019-11-10 10:56:42   
4  11640478 2019-11-03 16:19:00 2019-11-03 17:08:52 2019-11-03 17:08:52   

   Status                                       Status Notes  \
0    Open                                             closed   
1  Closed  This issue has already been reported through a...   
2  Closed                    See CAD# 193141269 for details.   
3  Closed                    See CAD# 193141258 for details.   
4  Closed                    See CAD# 193072861 for details.   

                 Responsible Agency           Category         Request Type  \
0  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   
1  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   
2  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   
3  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   
4  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   

       Request Details  ...     Neighborhood Police District          Y  \
0       homeless_other  ...  South of Market        SOUTHERN  37.774707   
1      wellbeing_check  ...   Lower Nob Hill        NORTHERN  37.788818   
2  aggressive_behavior  ...  Duboce Triangle            PARK  37.768701   
3       homeless_other  ...       Tenderloin      TENDERLOIN  37.784306   
4      wellbeing_check  ...          Mission         MISSION  37.755469   

            X                         Point          Source  Media URL  Year  \
0 -122.411510  (37.77470701, -122.41150998)  Mobile/Open311        NaN  2019   
1 -122.421570  (37.78881836, -122.42156982)           Phone        NaN  2019   
2 -122.428627  (37.76870057, -122.42862669)  Mobile/Open311        NaN  2019   
3 -122.416569  (37.78430598, -122.41656898)  Mobile/Open311        NaN  2019   
4 -122.418176  (37.75546882, -122.41817633)  Mobile/Open311        NaN  2019   

  Month  Day  
0    12    4  
1    11   21  
2    11   10  
3    11   10  
4    11    3  

[5 rows x 23 columns]

## Readability

In [17]:
df['Police District'] = df['Police District'].astype(str).str.lower() 
df['Supervisor District'] = df['Supervisor District'].astype(str).str.lower() 
df['Neighborhood'] = df['Neighborhood'].astype(str).str.lower() 
df['Address'] = df['Address'].astype(str).str.lower() 
df.head()

CaseID              Opened              Closed             Updated  \
0  11770932 2019-12-04 15:47:00                 NaT 2019-12-05 03:56:44   
1  11719142 2019-11-21 13:14:00 2019-11-21 14:08:41 2019-11-21 14:08:41   
2  11669592 2019-11-10 10:44:00 2019-11-10 10:56:28 2019-11-10 10:56:28   
3  11669528 2019-11-10 10:31:00 2019-11-10 10:56:42 2019-11-10 10:56:42   
4  11640478 2019-11-03 16:19:00 2019-11-03 17:08:52 2019-11-03 17:08:52   

   Status                                       Status Notes  \
0    Open                                             closed   
1  Closed  This issue has already been reported through a...   
2  Closed                    See CAD# 193141269 for details.   
3  Closed                    See CAD# 193141258 for details.   
4  Closed                    See CAD# 193072861 for details.   

                 Responsible Agency           Category         Request Type  \
0  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   
1  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   
2  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   
3  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   
4  Police - Homeless Concerns Queue  Homeless Concerns  Individual Concerns   

       Request Details  ...     Neighborhood Police District          Y  \
0       homeless_other  ...  south of market        southern  37.774707   
1      wellbeing_check  ...   lower nob hill        northern  37.788818   
2  aggressive_behavior  ...  duboce triangle            park  37.768701   
3       homeless_other  ...       tenderloin      tenderloin  37.784306   
4      wellbeing_check  ...          mission         mission  37.755469   

            X                         Point          Source  Media URL  Year  \
0 -122.411510  (37.77470701, -122.41150998)  Mobile/Open311        NaN  2019   
1 -122.421570  (37.78881836, -122.42156982)           Phone        NaN  2019   
2 -122.428627  (37.76870057, -122.42862669)  Mobile/Open311        NaN  2019   
3 -122.416569  (37.78430598, -122.41656898)  Mobile/Open311        NaN  2019   
4 -122.418176  (37.75546882, -122.41817633)  Mobile/Open311        NaN  2019   

  Month  Day  
0    12    4  
1    11   21  
2    11   10  
3    11   10  
4    11    3  

[5 rows x 23 columns]

In [18]:
print("Min date: ",min(df['Opened']))
print("Max date: ",max(df['Opened']))

Min date:  2015-10-08 09:51:05
Max date:  2019-12-04 15:47:00


In [20]:
df.to_csv('../../data/processed/homelessConcerns.csv')